routing/datasource.go

```diff
@@ -364,11 +364,15 @@ func processPredicates(cpm map[string]PredicateSpec, defs []*eskip.Predicate) ([
 	var weight int
 	for _, def := range defs {
 		if def.Name == predicates.WeightName {
+			var w int
 			var err error
-			if weight, err = parseWeightPredicateArgs(def.Args); err != nil {
+
+			if w, err = parseWeightPredicateArgs(def.Args); err != nil {
 				return nil, 0, err
 			}
 
+			weight += w
```

Comment:
This would sum all occurrences of `Weight(), so  `Weight(5) && Weight(15) = 20` and not `15` anymore

Reply:
Correct. And that would technically be a backwards incompatible change. But being honest, I don't think anyone would expect that `PredicateA() && PredicateB()` worked differently than `PredicateB() && PredicateA()`...

Reply:
I agree